examples/chat.rs

```diff
@@ -0,0 +1,175 @@
+        .message_id_fn(message_id_fn) // content-address messages. No two messages of the same content will be propagated.
+        .build()
+        .expect("Valid config");
+
+    // build a gossipsub network behaviour
+    let mut gossipsub = gossipsub::Behaviour::new(
+        gossipsub::MessageAuthenticity::Signed(id_keys),
+        gossipsub_config,
+    )
+    .expect("Correct configuration");
+    // Create a Gossipsub topic
+    let topic = gossipsub::IdentTopic::new("test-net");
+    // subscribes to our topic
+    gossipsub.subscribe(&topic)?;
+
+    let nym_id = rand::random::<u64>().to_string();
+    let docker_client = clients::Cli::default();
+    let (_nym_container, dialer_uri) = create_nym_client(&docker_client, &nym_id);
+    info!("dialer_uri: {}", dialer_uri);
+
+    let local_key = identity::Keypair::generate_ed25519();
+    let local_peer_id = PeerId::from(local_key.public());
+    info!("Local peer id: {local_peer_id:?}");
+
+    let transport = NymTransport::new(&dialer_uri, local_key).await?;
+
+    let mut swarm = SwarmBuilder::with_tokio_executor(
+        transport
+            .map(|a, _| (a.0, StreamMuxerBox::new(a.1)))
+            .boxed(),
+        Behaviour { gossipsub },
+        local_peer_id,
+    )
+    .build();
+
+    if let Some(addr) = std::env::args().nth(1) {
+        let remote: Multiaddr = addr.parse()?;
+        swarm.dial(remote)?;
+        info!("Dialed {addr}")
+    }
+
+    println!("swarm has been built");
+
+    // Read full lines from stdin
+    let mut stdin = codec::FramedRead::new(io::stdin(), codec::LinesCodec::new()).fuse();
+
+    println!("Enter messages via STDIN and they will be sent to connected peers using Gossipsub");
+
+    // Kick it off
+    loop {
+        select! {
+            line = stdin.select_next_some() => {
+                if let Err(e) = swarm
+                    .behaviour_mut().gossipsub
+                    .publish(topic.clone(), line.expect("Stdin not to close").as_bytes()) {
+                    println!("Publish error: {e:?}");
+                }
+            },
+            event = swarm.select_next_some() => {
+                println!("The event is: {:?}", event);
+                match event {
+                    SwarmEvent::Behaviour(BehaviourEvent::Gossipsub(gossipsub::Event::Message {
+                        propagation_source: peer_id,
+                        message_id: id,
+                        message,
+                    })) => println!(
+                            "Got message: '{}' with id: {id} from peer: {peer_id}",
+                            String::from_utf8_lossy(&message.data),
+                        ),
+                    SwarmEvent::NewListenAddr { address, .. } => {
+                        println!("Local node is listening on {address}");
+                    }
+                    other => {println!("other stuff: {:?}", other)}
```

Comment:
```suggestion
                    other => {println!("other event: {:?}", other)}
```

events that are "other" are logged twice